File: dart/rna-transcription/lib/rna_transcription.dart
class RnaTranscription {
		String toRna(String strand) {
			Map<String, String> rnaMap = {
				'G': 'C',
				'C': 'G',
				'T': 'A',
				'A': 'U'
			};
			String res = '';
			for (int i = 0; i < strand.length; i++) {
				if (!rnaMap.containsKey(strand[i])) {
					throw new ArgumentError('bad');
				}
				res += rnaMap[strand[i]];
			}
			return res;
		}
}

File: csharp/rna-transcription/RnaTranscription.cs
using System;
using System.Collections.Generic;
using System.Text;

public static class RnaTranscription
{
    private static Dictionary<char, char> map = new Dictionary<char, char> {
       { 'G', 'C' },
       { 'C', 'G' },
       { 'T', 'A' },
       { 'A', 'U' },
    };
    public static string ToRna(string nucleotide)
    {
        StringBuilder res = new StringBuilder(); 
        foreach (char c in nucleotide) {
            if (map.ContainsKey(c)) {
                res.Append(map[c]);
            }
        }
        return res.ToString();
    }
}